sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1040,3 +1044,57 @@ class JoinSuite extends QueryTest with SharedSparkSession {
     checkAnswer(df, Row(1, 2, 1, 2) :: Nil)
   }
 }
+
+class JoinWithResourceCleanSuite extends JoinSuite with BeforeAndAfterAll {
+  import testImplicits._
+  import scala.collection.mutable.ArrayBuffer
+
+  private def checkCleanupResourceTriggered(plan: SparkPlan): ArrayBuffer[SortExec] = {
```

Comment:
It took me a while to understand the tests.

So this method doesn't trigger the check, but attach the checker which will be run later.

How about we redefine this method as `def attachChecker(checkCodegen: Boolean): Unit`?

Reply:
Sorry for the mess, after we delete the config, this Suite can be removed, done in 7787d45.